codec-classes-quic/src/main/java/io/netty/handler/codec/quic/QuicheQuicStreamChannel.java

```
@@ -485,6 +381,68 @@ public RecvByteBufAllocator.Handle recvBufAllocHandle() {
             return recvHandle;
         }
 
+        @Override
+        public void shutdown(ChannelShutdownType type, ChannelPromise promise) {
+            if (type.data() != null && !(type.data() instanceof Integer)) {
+                promise.setFailure(new IllegalArgumentException(
+                        "ChannelShutdownType with data if non integer type is allowed: " + type));
+                return;
+            }
+            final boolean read;
+            final boolean write;
+            switch (type.direction()) {
+                case Outbound:
+                    write = true;
+                    read = false;
+                    break;
+                case Inbound:
+                    read = true;
+                    write = false;
+                    break;
+                default:
+                    promise.setFailure(new UnsupportedOperationException());
+                    return;
+            }
+
+            if (write) {
+                ioTransport.writeWithoutCheckChannelState(QuicStreamFrame.EMPTY_FIN, newPromise()
```

Should we skip this and fail early if `outputShutdown` is already `true`? Similar for inbound.

---

we could ... I just ported the old code.

---

I think we should not fail but just succeed them early. As even on a idk SocketChannel it will not throw in this case. ... done